### PR TITLE
Hive + MariaDb

### PR DESCRIPTION
**Pull request by imakhlin**
See also PR #15

### CODE REVIEW

File: services/hive/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+ARG DOCKER_USER
+ARG HADOOP_VERSION=2.9.0
+ARG HIVE_VERSION=2.3.2
```
Comment (bivas) [inline services/hive/docker/Dockerfile]: Don't default these args. We wish to have them known (of fail) during build

File: services/mariadb/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+ARG MARIADB_VERSION=10.4
```
Comment (bivas) [inline services/mariadb/docker/Dockerfile]: Don't default

File: services/mariadb/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      /etc/mysql \
+      /usr/lib/mysql/plugin/auth_pam_tool_dir
+
+USER $MARIADB_UID
+
+ARG MYSQL_ROOT_PASSWORD=ChangeMe!
+ENV MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
+ENV HOME /home/$MARIADB_UNAME
+
+RUN set -x \
```
Comment (bivas) [inline services/mariadb/docker/Dockerfile]: We usually add these instruction in the chart.

File: services/mariadb/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    && apt-get install -yq --no-install-recommends curl \
+    && mkdir -p /var/lib/mysql/mysql /docker-entrypoint-initdb.d \
+    && chown -R $MARIADB_UID:$MARIADB_GID /home/$MARIADB_UNAME \
+      /var/lib/mysql \
+      /etc/mysql \
+      /usr/lib/mysql/plugin/auth_pam_tool_dir
+
+USER $MARIADB_UID
+
+ARG MYSQL_ROOT_PASSWORD=ChangeMe!
```
Comment (bivas) [inline services/mariadb/docker/Dockerfile]: ARG and ENV should be before instructions.

File: services/mariadb/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ENV MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
+ENV HOME /home/$MARIADB_UNAME
```
Comment (bivas) [inline services/mariadb/docker/Dockerfile]: Don't call it `HOME` but `MARIADB_HOME` (home is shell specific)

File: services/mariadb/docker/Dockerfile (hunk 3 above)
Comment (imakhlin) [reply services/mariadb/docker/Dockerfile]: are you talking about `MYSQL_ROOT_PASSWORD` ?

Comment (bivas) [reply services/mariadb/docker/Dockerfile]: Aliases

File: services/mariadb/docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+FROM mariadb:${MARIADB_VERSION}
+
+# Note, base image assumes that "mariadb" user have UID/GID=1001
+ARG MARIADB_UNAME=mysql
+ARG MARIADB_UID=1001
+ARG MARIADB_GID=1001
+
+ARG MYSQL_ROOT_PASSWORD=ChangeMe!
+ENV MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
```
Comment (bivas) [inline services/mariadb/docker/Dockerfile]: Remove these `ARG` and `ENV` - should be assigned by the deployment

**Pull request merged**